src/controller.cpp

```diff
@@ -1135,32 +1137,54 @@ void controller::enqueue_items(std::shared_ptr<rss_feed> feed)
 }
 
 std::string controller::generate_enqueue_filename(const std::string& url,
+	const std::string& title,
+	const time_t pubDate,
 	std::shared_ptr<rss_feed> feed)
 {
 	std::string dlformat = cfg.get_configvalue("download-path");
 	if (dlformat[dlformat.length() - 1] != NEWSBEUTER_PATH_SEP[0])
 		dlformat.append(NEWSBEUTER_PATH_SEP);
 
-	fmtstr_formatter fmt;
-	fmt.register_fmt('n', feed->title());
-	fmt.register_fmt('h', get_hostname_from_url(url));
+	std::string filemask = cfg.get_configvalue("download-filename");
+	dlformat.append(filemask);
 
-	std::string dlpath = fmt.do_format(dlformat);
+	auto time_formatter = [&pubDate](const char* format) {
+		char pubDate_formatted[1024];
+		strftime(pubDate_formatted,
+				sizeof(pubDate_formatted),
+				format,
+				localtime(&pubDate));
+		return  std::string(pubDate_formatted);
+	};
 
 	char buf[2048];
 	snprintf(buf, sizeof(buf), "%s", url.c_str());
 	char* base = basename(buf);
-	if (!base || strlen(base) == 0) {
-		char lbuf[128];
-		time_t t = time(nullptr);
-		strftime(lbuf,
-			sizeof(lbuf),
-			"%Y-%b-%d-%H%M%S.unknown",
-			localtime(&t));
-		dlpath.append(lbuf);
-	} else {
-		dlpath.append(base);
+	std::string extension;
+	if (base) {
+		std::string base_s (base);
+		std::size_t pos = base_s.rfind('.');
+		if (pos != std::string::npos)
+			extension.append(base_s.substr(pos+1));
 	}
+
+	fmtstr_formatter fmt;
+	fmt.register_fmt('n', feed->title());
+	fmt.register_fmt('h', get_hostname_from_url(url));
+	fmt.register_fmt('u', base);
+	fmt.register_fmt('F', time_formatter(_("%F")));
+	fmt.register_fmt('m', time_formatter(_("%m")));
+	fmt.register_fmt('b', time_formatter(_("%b")));
+	fmt.register_fmt('d', time_formatter(_("%d")));
+	fmt.register_fmt('H', time_formatter(_("%H")));
+	fmt.register_fmt('M', time_formatter(_("%M")));
+	fmt.register_fmt('S', time_formatter(_("%S")));
+	fmt.register_fmt('y', time_formatter(_("%y")));
+	fmt.register_fmt('Y', time_formatter(_("%Y")));
+	fmt.register_fmt('t', title);
+	fmt.register_fmt('e', extension);
+
+	std::string dlpath = fmt.do_format(dlformat);
```

Comment:
This function combines "download-path" and "download-filename" settings into `dlformat` and then performs the formatting. As a result, `%n` and `%h` would work in "download-filename", and other formats will work in "download-path".

This actually introduces an opportunity for breakage: what if some user has date and/or time placeholders in their "download-path", and expect them to *not* be replaced? I believe the benefit outweighs the risk here, so let's proceed; but placeholders need to be documented. I suggest adding a new table to "Format Strings" section of the doc (doc/newsboat.txt:919), and linking to it from descriptions in doc/podboat-cmds.dsv.

Reply:
Should I document all of `strftime` format options as well.

Reply:
Yes, if you don't mind.

Reply:
I like the changes you've made in recent commits, but they still keep e.g. `%n` and `%t` separate. When I wrote about table, I meant a *single* table for both `download-path` and `download-filename-format`, since they both use the same placeholders. You can also remove placeholders from doc/podboat-cmds.dsv, since we'll have a table now.

I feel the review is dragging out a bit. If you prefer, I can merge this PR as-is—it's near-perfect IMHO—and fixing the docs myself later. Should I?

Reply:
Can you check it out now.

Reply:
That's perfect! I'll merge once Travis finishes the build.